transcom/mymove
# [MB-15876] Prime API endpoint to create a SIT address change request

## [Jira ticket](tbd)

## Summary

This ticket creates the new endpoint `/sit-address-updates` and handler to allow the `PRIME` to create a SIT address update `REQUEST`. 

The status is set based on distance, <= 50 miles results in an APPROVED status while anything beyond that results in a REQUESTED status. This was done due to the conversation in [this thread](https://ustcdp3.slack.com/archives/C04U4TWDLKG/p1684533154848419?thread_ts=1684530850.656589&cid=C04U4TWDLKG), however it is assumed that this new endpoint will be used for requests greater than 50 miles.

When an approval request is successfully created the move status will also be updated to `APPROVALS_REQUESTED` ([See this thread](https://ustcdp3.slack.com/archives/CTNTFJSBA/p1684446431717639))

You will see some `ghcapi` changes in this PR as well which come from me pulling in the changes [in this branch](https://github.com/transcom/mymove/pull/10698) since it involved some changes needed for this PR as well.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test
1. Run `make server_test` and the new tests should pass

Suggested Verification approaches: 

>**Note** The expected body params for `/sit-address-updates` are: `ContractorRemarks`, `MtoServiceItemID`, and `NewAddress`

1. [HTTP](https://github.com/transcom/mymove/pull/10698#pullrequestreview-1437072374)
3. [Postman](https://github.com/transcom/mymove/pull/10698#issue-1716000662)

### Backend
